(RHEL-30372) efi: check if all sections of our EFI binaries are properly aligned

mrc0mmand opened the pull request:
(cherry picked from commit 7ff3b88396f440bff60328b4bff7627a34d45e4a)

Related: TBD

<!-- issue-commentator = {"comment-id":"2018472309"} -->

keszybz committed:
ci: install pefile

(cherry picked from commit 1f6da5d902ef192c7fe24ffd4ebb3b9e1f2ecda8)

Related: RHEL-30372

mrc0mmand committed:
efi: check if all sections of our EFI binaries are properly aligned

(cherry picked from commit 7ff3b88396f440bff60328b4bff7627a34d45e4a)

Related: RHEL-30372

jamacku reviewed (commented):
@mrc0mmand You can now reference [RHEL-30372](https://issues.redhat.com/browse/RHEL-30372)

aafeijoo-suse committed:
efi: skip check-alignment-* tests if pefile is not installed

Otherwise, when building with ukify disabled, no one else requires pefile, and
if it is not installed, those tests fail:

```
================================== 12/1212 ===================================
test:         systemd:efi / check-alignment-systemd-bootx64.efi
start time:   14:21:54
duration:     0.08s
result:       exit status 1
command:      /mnt/work/systemd/upstream-fork/main/tools/check-efi-alignment.py /mnt/work/systemd/upstream-fork/main/build/src/boot/efi/systemd-bootx64.efi
----------------------------------- stderr -----------------------------------
Traceback (most recent call last):
  File "/mnt/work/systemd/upstream-fork/main/tools/check-efi-alignment.py", line 7, in <module>
    import pefile
ModuleNotFoundError: No module named 'pefile'
==============================================================================

================================== 13/1212 ===================================
test:         systemd:efi / check-alignment-linuxx64.efi.stub
start time:   14:21:54
duration:     0.08s
result:       exit status 1
command:      /mnt/work/systemd/upstream-fork/main/tools/check-efi-alignment.py /mnt/work/systemd/upstream-fork/main/build/src/boot/efi/linuxx64.efi.stub
----------------------------------- stderr -----------------------------------
Traceback (most recent call last):
  File "/mnt/work/systemd/upstream-fork/main/tools/check-efi-alignment.py", line 7, in <module>
    import pefile
ModuleNotFoundError: No module named 'pefile'
==============================================================================

================================== 14/1212 ===================================
test:         systemd:efi / check-alignment-addonx64.efi.stub
start time:   14:21:54
duration:     0.07s
result:       exit status 1
command:      /mnt/work/systemd/upstream-fork/main/tools/check-efi-alignment.py /mnt/work/systemd/upstream-fork/main/build/src/boot/efi/addonx64.efi.stub
----------------------------------- stderr -----------------------------------
Traceback (most recent call last):
  File "/mnt/work/systemd/upstream-fork/main/tools/check-efi-alignment.py", line 7, in <module>
    import pefile
ModuleNotFoundError: No module named 'pefile'
==============================================================================
```

Follow-up for 7ff3b88396f440bff60328b4bff7627a34d45e4a

(cherry picked from commit f0896ca55781c842996b018b35ba63917d8af444)

Related: RHEL-30372

mrc0mmand commented:
Updated the references and marked for review, even though it'll keep failing until the undrelying objcopy issue is fixed (should be in 9.5).

Also, it's very possible we might ditch the gnu-efi+objcopy approach completely in the meanwhile due to https://issues.redhat.com/browse/RHEL-30400.
